packages/ERTP/src/issuer.chainmail

```diff
@@ -0,0 +1,213 @@
+/**
+ * The issuer cannot mint new amount, but it can create empty purses and
+ * payments. The issuer can also transform payments (splitting payments,
+ * combining payments, burning payments, and claiming payments
+ * exclusively). The issuer should be gotten from a trusted source and
+ * then relied upon as the decider of whether an untrusted payment is valid. 
+ */
+interface Issuer (Amount (Extent)) {
+  /**
+   * Get the Brand for this Issuer. The Brand indicates the kind of
+   * digital asset and is shared by the mint, the issuer, and any purses
+   * and payments of this particular kind. The brand is not closely
+   * held, so this function should not be trusted to identify an issuer
+   * alone. Fake digital assets and amount can use another issuer's brand.
+   */
+  getBrand() -> (Brand);
+
+  /* Get the allegedName for this mint/issuer */
+  allegedName() -> (allegedName);
+
+  /* Get the AmountMath for this Issuer. */
+  getAmountMath() -> (AmountMath);
+
+  /* Get the name of the MathHelpers for this Issuer. */
+  getMathHelpersName() -> (String);
+
+  /** Make an empty purse of this brand. */
+  makeEmptyPurse(memo :String) -> (Purse);
+
+  /**
+   * Get payment balance. Because the payment is not trusted, we
+   * cannot call a method on it directly, and must use the issuer
+   * instead.
+   */ 
+  getBalance(payment :Payment) -> (Amount);
```

Comment:
Something's been itching for a while; i'm sharing it now because it feels relevant
`getBalance` makes sense for the natAmountMath type mints/issuers, but not so much for non-fungible rights. `getAmount` might make more sense in general

Reply:
I had not thought about the difference between purse and payment (and realize now i had only commented on the payment part and that there is still a `getBalance` on purse)

> for purses, getAmount is worse than getBalance because the term should emphasize that the amount in the purse changes over time and what you're getting is a snapshot of the current amount.

My impression is that `purse.getAmount()` follows the programming naming conventions that already conveys that "what you're getting is a snapshot of the current amount". It is a method call and the convention for method calls is that they return a value that is correct at the time of call (and the returned value may change if the same method is called at a different time)
Like when using a standard Map, `myMap.get('foo')` returns the value at the time of call and the returned value may change if the same method is called at a different time

If a difference really needs to be made, `getCurrentAmount` might work for the purse case

As a side note, i find interesting that the example you gave is "my bank account has a balance". A "purse" does not have a balance i think (and i anticipate why Purse is not named "Account"). I feel "balance" is borrowed from one metaphor and glued onto another

Reply:
`getAmount` for payments and `getCurrentAmount` for purses helps both to emphasize their similarities and their differences. A "current amount" is clearly a kind of amount; whereas a "balance" might be another thing until you learn that it is an amount. I think I like it.

Reply:
I'm a big fan of reducing the number of terms and concepts, so I really like the `getAmount` idea. @erights, I agree with David here that `getAmount` already conveys it's a snapshot, rather than guaranteeing anything about the value of future calls being the same. I could also see it being cumbersome for the user to remember which method they are supposed to use. Could we talk this through during this week's meeting?